src/Surfnet/StepupRa/RaBundle/Service/AuditLogService.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupRa\RaBundle\Service;
+
+use Surfnet\StepupMiddlewareClient\Identity\Dto\SecondFactorAuditLogSearchQuery;
+use Surfnet\StepupMiddlewareClientBundle\Identity\Dto\AuditLog;
+use Surfnet\StepupMiddlewareClientBundle\Identity\Service\AuditLogService as ApiAuditLogService;
+use Surfnet\StepupRa\RaBundle\Command\SearchSecondFactorAuditLogCommand;
+
+class AuditLogService
+{
+    /**
+     * @var ApiAuditLogService
+     */
+    private $apiAuditLogService;
+
+    public function __construct(ApiAuditLogService $auditLogService)
+    {
+        $this->apiAuditLogService = $auditLogService;
+    }
+
+    /**
+     * @param SearchSecondFactorAuditLogCommand $command
+     * @return AuditLog
+     */
+    public function getAuditlog(SearchSecondFactorAuditLogCommand $command)
+    {
+        $query = new SecondFactorAuditLogSearchQuery($command->institution, $command->identityId, $command->pageNumber);
+        $query->setOrderBy($command->orderBy);
+        $query->setOrderDirection($command->orderDirection);
+
+        $result = $this->apiAuditLogService->searchSecondFactorAuditLog($query);
+
+        if ($result === null) {
+            return new AuditLog([], 0, 1, 25);
```

Comment:
Should not happen :tm: Doesn't mean it won't and at this point in the code we should not assume that an API always behaves as expected (think about borked releases, people being stupid, servers that throw out 404's, etc). That is abstracted in the client+bundle, which in case of 404 gracefully returns `null` so that the consumer of the client+bundle can choose how to resolve. 

As far as this service is concerned, it either receives data, or there is no data in which case (for ease of handling at the controller level) it just returns an empty data set. Whatever the cause of the "no data" (`null`) is not relevant to this service, but it should be able to handle it as it is a valid, documented return value of the `AuditLogService`.